core/src/main/scala/com/github/mrpowers/spark/fast/tests/ProductUtil.scala

```diff
@@ -1,35 +1,50 @@
 package com.github.mrpowers.spark.fast.tests
 
 import com.github.mrpowers.spark.fast.tests.ufansi.Color.{DarkGray, Green, Red}
+import com.github.mrpowers.spark.fast.tests.ufansi.FansiExtensions.StrOps
 import org.apache.commons.lang3.StringUtils
 import org.apache.spark.sql.Row
-import com.github.mrpowers.spark.fast.tests.ufansi.FansiExtensions.StrOps
-object DataframeUtil {
 
-  private[mrpowers] def showDataframeDiff(
+import scala.reflect.ClassTag
+
+object ProductUtil {
+  private[mrpowers] def productOrRowToSeq(product: Any): Seq[Any] = {
+    product match {
+      case null       => Seq.empty
+      case r: Row     => r.toSeq
+      case p: Product => p.productIterator.toSeq
+      case _          => throw new IllegalArgumentException("Only Row and Product types are supported")
+    }
+  }
+  private[mrpowers] def showProductDiff[T: ClassTag](
       header: (String, String),
-      actual: Seq[Row],
-      expected: Seq[Row],
+      actual: Seq[T],
+      expected: Seq[T],
       truncate: Int = 20,
-      minColWidth: Int = 3
+      minColWidth: Int = 3,
+      defaultVal: T = null.asInstanceOf[T]
   ): String = {
 
+    val runTimeClass                     = implicitly[ClassTag[T]].runtimeClass
+    val className                        = runTimeClass.getSimpleName
```

Comment:
@SemyonSinchenko @alfonsorr Do we want to display class name similar to Chispa or it's fine to omit that information

Reply:
Yes it will. 
I am not sure which style is Chispa having since the doc is showing two different result
![image](https://github.com/user-attachments/assets/3c954812-b123-4322-90e6-33390ee170bb)

Should we go with having a display everything as `ClassName(value1, value2, ...)`

Reply:
I have reverted so Row will not display any ClassName